tests/Aspire.Hosting.Azure.Tests/AzureEventHubsExtensionsTests.cs

```diff
@@ -583,4 +584,73 @@ public void AzureEventHubsAppliesAzureFunctionsConfiguration()
         Assert.Equal("cg1", target["Aspire__Azure__Messaging__EventHubs__EventHubBufferedProducerClient__cg1__ConsumerGroup"]);
         Assert.Equal("hub1", target["Aspire__Azure__Messaging__EventHubs__EventHubBufferedProducerClient__cg1__EventHubName"]);
     }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AzureEventHubs_WithPersistentLifetime_ReusesContainers()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        using var builder1 = TestDistributedApplicationBuilder.Create(testOutputHelper);
+
+        // We don't need to clean on StopAsync(), we just check that the new instances are not new ones
+        builder1.Configuration["DcpPublisher:WaitForResourceCleanup"] = "false";
+
+        var resource = builder1.AddAzureEventHubs("resource")
+                      .RunAsEmulator(x => x.WithLifetime(ContainerLifetime.Persistent));
+
+        resource.AddHub("hub");
+
+        using var app = builder1.Build();
+
+        await app.StartAsync(cts.Token);
+
+        var rns = app.Services.GetRequiredService<ResourceNotificationService>();
+
+        await rns.WaitForResourceHealthyAsync(resource.Resource.Name, cts.Token);
+
+        var resourceEvent = await rns.WaitForResourceAsync("resource", e => GetContainerId(e) is not null, cts.Token);
+        var ehContainerId1 = GetContainerId(resourceEvent);
+
+        // NOTE: Waiting for Running state doesn't guaranty that the container-id property is set (for this resource at least)
+
+        resourceEvent = await rns.WaitForResourceAsync("resource-storage", e => GetContainerId(e) is not null, cts.Token);
```

Comment:
@karolz-ms 

I had some problems locally with this. I am trying to read the `"container-id"` property of the snapshot. However if I wait for the Running state and then read the id right after it returns null most of the time. For the previous resource (a parent) it works all the time. So now I use the actual value as the predicate. Also these containers were already running on my machine (persistent from a previous run) so the container always has an id and is running.

Reply:
It is a little surprising that a container in Running state does not have a container ID set since on DCP side this is done as one change that should be atomic.  But what you are doing here certainly makes sense.

The other thing to keep in mind is that even for running containers, the sequence of events represents the lifetime of the Container object in DCP. So the first event will always be a "Created" event with no container ID, and then DCP will notice that the existing Docker container can be reused, so it will fill the ContainerID property and change its Container object state to Running. And that change will be delivered as a separate event.